Sources/MapboxMaps/Gestures/GestureHandlers/PanGestureHandler.swift

```diff
@@ -42,31 +56,46 @@ internal final class PanGestureHandler: GestureHandler {
             cameraAnimationsManager.cancelAnimations()
             delegate?.gestureBegan(for: .pan)
         case .changed:
+            guard let initialTouchLocation = initialTouchLocation,
+                  let initialCameraState = initialCameraState else {
+                return
+            }
             lastChangedDate = dateProvider.now
             cameraAnimationsManager.cancelAnimations()
-            handleChange(withTouchLocation: touchLocation)
+            handleChange(
+                withTouchLocation: touchLocation,
+                initialTouchLocation: initialTouchLocation,
+                initialCameraState: initialCameraState)
         case .ended:
             // Only decelerate if the gesture ended quickly. Otherwise,
             // you get a deceleration in situations where you drag, then
             // hold the touch in place for several seconds, then release
             // it without further dragging. This specific time interval
             // is just the result of manual tuning.
             let decelerationTimeout: TimeInterval = 1.0 / 30.0
-            guard let lastChangedDate = lastChangedDate,
-                  dateProvider.now.timeIntervalSince(lastChangedDate) < decelerationTimeout,
-                  let decelerationRate = delegate?.decelerationRate else {
+            guard let initialTouchLocation = initialTouchLocation,
+                  let initialCameraState = initialCameraState,
+                  let lastChangedDate = lastChangedDate,
+                  dateProvider.now.timeIntervalSince(lastChangedDate) < decelerationTimeout else {
                 return
             }
             cameraAnimationsManager.decelerate(
                 location: touchLocation,
                 velocity: gestureRecognizer.velocity(in: view),
-                decelerationRate: decelerationRate,
-                locationChangeHandler: handleChange(withTouchLocation:),
+                decelerationFactor: decelerationFactor,
+                locationChangeHandler: { (touchLocation) in
+                    // here we capture the initial state so that we can clear
+                    // it immediately after starting the animation
+                    self.handleChange(
+                        withTouchLocation: touchLocation,
+                        initialTouchLocation: initialTouchLocation,
+                        initialCameraState: initialCameraState)
+                },
                 completion: {
-                    self.initialTouchLocation = nil
-                    self.initialCameraState = nil
-                    self.lastChangedDate = nil
```

Comment:
Should we make this completion optional? Seems like it should be nullable instead of sending an empty closure here.

Reply:
let's address this in a future pr.